src/components/App/AppIFrame.js

```diff
@@ -84,20 +73,31 @@ class AppIFrame extends React.Component {
       // Also navigate when it's the same app, but a different instance
       nextApp.proxyAddress !== this.props.app.proxyAddress
     ) {
-      this.resetProgress(() => {
-        this.navigateIFrame(nextApp.src)
-      })
+      this.navigateIFrame(nextApp.src)
     }
   }
   componentWillUnmount() {
-    this.identitySubscription.unsubscribe()
     window.removeEventListener('message', this.handleReceiveMessage, false)
-    this.clearProgressTimeout()
+    if (this.state.isLoading) {
+      this.props.onLoadingCancel()
+    }
+  }
+  loadingStart() {
+    if (this.state.isLoading) {
+      this.props.onLoadingCancel()
```

Comment:
Is this in case another app was loading?

Reply:
Yes :-)